jax/_src/core.py

```diff
@@ -3025,9 +3025,11 @@ class JaxprPpSettings(NamedTuple):
 # Jaxprs.
 class JaxprPpContext:
   var_ids: DefaultDict[Var, int]
+  seen_jaxprs: set[int]
```

Comment:
Don't jaxprs hash on id anyway, so this could be a `set[Jaxpr]`, and we wouldn't need to call `id`?

Reply:
You are right, updated.